tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/BrokerRequestTests.cs

```diff
@@ -81,6 +81,50 @@ public void BrokerErrorResponseTest()
                 });
         }
 
+        [TestMethod]
+        public void BrokerInteractionRequiredErrorResponseTest()
+        {
+            CreateBrokerHelper();
+
+            var response = new MsalTokenResponse
+            {
+                Error = MsalError.InteractionRequired,
+                ErrorDescription = MsalError.InteractionRequired
+            };
+
+            ValidateBrokerResponse(
+                response,
+                exception =>
+                {
+                    var exc = exception as MsalUiRequiredException;
+                    Assert.IsNotNull(exc);
+                    Assert.AreEqual(MsalError.BrokerResponseReturnedError, exc.ErrorCode);
+                    Assert.AreEqual(MsalErrorMessage.BrokerResponseReturnedError, exc.Message);
+                });
+        }
+
+        [TestMethod]
+        public void BrokerInvalidGrantErrorResponseTest()
+        {
+            CreateBrokerHelper();
+
+            var response = new MsalTokenResponse
+            {
+                Error = MsalError.InvalidGrantError,
+                ErrorDescription = MsalError.InvalidGrantError
+            };
+
+            ValidateBrokerResponse(
```

Comment:
Please update validation code to check for UiRequiredException and for other exception details like the http stuff.

Reply:
This already checks for uirequiredexceptions via the callback. adding other http stuff